### PR TITLE
WebUI: Add support for custom table pagination size

### PR DESCRIPTION
**Pull request by pvomacka**
New customization button opens dialog with field for setting the number of lines
in tables. After saving the new value there is new topic which starts refreshing
current table facet (if shown) and set all other facets expired. Therefore all
tables are immediately regenerated.

https://fedorahosted.org/freeipa/ticket/5742

### CODE REVIEW

File: install/ui/src/freeipa/facet.js [hunk 1]
```diff
@@ -2359,6 +2361,14 @@ exp.table_facet = IPA.table_facet = function(spec, no_init) {
             selectable: that.selectable && !that.read_only
         });
 
+        topic.subscribe("change-pagination", function() {
+            that.table.page_length = config.get('table_page_size');
+
+            if (that.is_shown) that.refresh();
+
+            that.set_expired_flag();
```
Comment (pvoborni) [inline install/ui/src/freeipa/facet.js]: Should the expiration flag be set first and then issuing the refresh?

File: install/ui/src/freeipa/field.js [hunk 2]
```diff
@@ -962,6 +962,39 @@ field.validator = IPA.validator = function(spec) {
 };
 
 /**
+ * Javascript integer validator
+ *
+ * It allows to insert only integer numbers which can be safely represented by
+ * Javascript.
+ *
+ * @class
+ * @alternateClassName IPA.metadata_validator
```
Comment (pvoborni) [inline install/ui/src/freeipa/field.js]: alternate name is IPA.integer_validator

File: ipaserver/plugins/internal.py [hunk 3]
```diff
@@ -250,6 +250,10 @@ class i18n_messages(Command):
             "update": _("Update"),
             "view": _("View"),
         },
+        "customization": {
+            "customization": _("Customization"),
+            "table_pagination": _("Table Pagination Size"),
```
Comment (pvoborni) [inline ipaserver/plugins/internal.py]: What about using only "Pagination size". "Table" won't tell much to users. I.e. "what table?"

**Comment by pvomacka** [comment]
@pvoborni Thank you for review. Proposed changes fixed.

**Comment by martbab** [comment]
master:

* 7b699105a52d4d8c26a73044ba182d752b4a9833 Add javascript integer validator
* f78cc8932626de667c6e3a4461141a10a5d9c2e6 Make singleton from config module
* e1dfc51e48050ac1ad431d56003dc26e17ca653e Add support for custom table pagination size

**Pull request closed without merging**